src/debug_utilities/cn_deserialize.cpp

```diff
@@ -153,7 +153,11 @@ int main(int argc, char* argv[])
     std::cout << "Parsed transaction:" << std::endl;
     std::cout << cryptonote::obj_to_json_str(tx) << std::endl;
 
-    if (cryptonote::parse_tx_extra(tx.extra, fields))
+    bool parsed = cryptonote::parse_tx_extra(tx.extra, fields);
+    if (!parsed)
+      std::cout << "Failed to parse tx_extra" << std::endl;
+
+    if (!fields.empty())
```

Comment:
Shouldn't this be `else if`? Otherwise the failed-to-parse message and the no-field message are always displayed together, which is OK but seems a bit odd to me.

Reply:
If parsing fails, it will still have filled the returned vector with the fields it parsed before the error, if any. The point of this patch is to display those, even in the event of failure.

Reply:
Ah, got it. Thanks.